### PR TITLE
ELPP-3447 504 on timeout from ElasticSearch

### PR DESCRIPTION
**Pull request by giorgiosironi**
Not using `api-problem` yet, but it can be added later if it fits.

### CODE REVIEW

File: src/Search/Api/SearchController.php [hunk 1]
```diff
@@ -134,7 +136,11 @@ public function indexAction(Request $request, Accept $type)
                 break;
         }
 
-        $data = $this->client->searchDocuments($query->getRawQuery());
+        try {
+            $data = $this->client->searchDocuments($query->getRawQuery());
+        } catch (NoNodesAvailableException $e) {
```
Comment (thewilkybarkid) [inline src/Search/Api/SearchController.php]: Does this wrap timeout errors from cURL?

Comment (giorgiosironi) [reply src/Search/Api/SearchController.php]: It is what comes out from the black box ElasticSearch SDK (see https://elifesciences.atlassian.net/browse/ELPP-3447?focusedCommentId=30254&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-30254) as it doesn't expose the underlying requests and implements some (unused so far) retry logic.

Comment (thewilkybarkid) [reply src/Search/Api/SearchController.php]: We can handle things as we see them. Was trying to understand how it works a bit (looks to unfortunately use RingPHP+cURL).

Comment (giorgiosironi) [reply src/Search/Api/SearchController.php]: Uhm I wonder which of the twelve Guzzle major versions that corresponds to

Comment (thewilkybarkid) [reply src/Search/Api/SearchController.php]: Part of the short-lived Guzzle 5.

**Comment by giorgiosironi** [comment]
Got this also in https://alfred.elifesciences.org/job/test-journal-cms/428/consoleFull

**Pull request merged**